compiler/aarch64/codegen/ARM64OutOfLineCodeSection.cpp

```diff
@@ -39,5 +41,31 @@ void TR_ARM64OutOfLineCodeSection::assignRegisters(TR_RegisterKinds kindsToBeAss
 
 void TR_ARM64OutOfLineCodeSection::generateARM64OutOfLineCodeSectionDispatch()
    {
-   TR_UNIMPLEMENTED();
+   // Switch to cold helper instruction stream.
+   //
+   swapInstructionListsWithCompilation();
+
+   generateLabelInstruction(_cg, TR::InstOpCode::label, _callNode, _entryLabel);
+
+   TR::Register *resultReg = NULL;
+   if (_callNode->getOpCode().isCallIndirect())
+      resultReg = TR::TreeEvaluator::performCall(_callNode, true, _cg);
+   else
+      resultReg = TR::TreeEvaluator::performCall(_callNode, false, _cg);
+
+   if (_targetReg)
+      {
+      TR_ASSERT(resultReg, "assertion failure");
```

Comment:
Please provide a more descriptive assert message.

Reply:
Rewrote the message.